src/main/java/com/amazon/opendistroforelasticsearch/security/filter/OpenDistroSecurityFilter.java

```diff
@@ -169,7 +171,18 @@ public int order() {
                 attachSourceFieldContext(request);
             }
             final Set<String> injectedRoles = rolesInjector.injectUserAndRoles(threadContext);
-            final User user = threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER);
+            boolean isInjectedUser = false;
+            User user = threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER);
+            if(user == null && threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_INJECTED_USER) != null) {
+                String principal = threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_SSL_TRANSPORT_PRINCIPAL);
+                User injectedUser = backendRegistry.authenticate(request, principal, task, action);
```

Comment:
`authenticate()` may return `principal` instead of `ConfigConstants.OPENDISTRO_SECURITY_INJECTED_USER`. Is it expected? If yes, why is it necessary to inject `principal`? Also in this case please rename `injectedUser` and `isInjectedUser` to reflect that it can be SSL principal user.

Reply:
Made necessary changes
 - Didn't pass principal as only user injection is needed here.
 - Renamed the variable to enforcePrivilegesEvaluation